src/main/kotlin/com/sourcegraph/utils/CodyEditorUtil.kt

```diff
@@ -215,18 +198,25 @@ object CodyEditorUtil {
     }
   }
 
-  fun createFileOrScratch(
+  fun createFileOrScratchFromUntitled(
       project: Project,
       uriString: String,
       content: String? = null
   ): VirtualFile? {
     try {
-      val uri = URI.create(uriString).withScheme("file")
-      if (!uri.toPath().exists()) {
-        uri.toPath().parent?.createDirectories()
-        uri.toPath().createFile()
+      val uri = URI.create(uriString)
+
+      // For the sake of correctness we only allow untitled files to be created.
+      // If we receive a URI with a scheme other than "untitled" it should already exist on the disk
+      if (uri.scheme != "untitled") return null
```

Comment:
I do not like this return statement and the fact we need the comment. If this method assumes the file does not exist we should simply throw an exception and handle it higher. 

Or maybe we could just do this check where this method is called - in `src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentService.kt`?

Reply:
Returning null is consistent with other IJ file APIs where if file cannot be created it returns a `null` (e.g. `ScratchRootType::createScratchFile`)
As for comment I'm open for discussion.
It is not `needed` to understand this function, but I think it's a domain knowledge detail to understand that `file://` coming from VSC is almost certainly existing file while `untitiled://` is not. And as such it might deserve a comment, perhaps on the top of the function to document it.

Reply:
if it's consistent with IJ let it be as it is 🆗